src/App.js

```diff
@@ -45,6 +71,70 @@ const App = () => {
     //    3 squares in each column match
     // 3. Go across each diagonal to see if
     //    all three squares have the same value.
+    let hasEmpty = false;
```

Comment:
Nice work handling rows, columns, and diagonals, and using loops to DRY up your code. Your use of temp variables makes it clear how you broke down the problem. You may consider refactoring to remove some of these temp variables (i.e. `firstSquare`, `firstRow`)